tests/js/client/shell/shell-api-batch.js

```diff
@@ -0,0 +1,741 @@
+        let m = r.exec(s);
+        contentId = m[m.length - 1];
+      } else {
+        contentId = "";
+      }
+      // strip envelope;
+      partTotal = partTotal.slice(position + 4, partTotal.length);
+      // look for actual header & body
+      let partHeader = "";
+      let partBody = "";
+      position = partTotal.search("\r\n\r\n");
+
+      if (position === undefined) {
+        break;
+      }
+
+      partHeader = partTotal.slice(0, position);
+      partBody = partTotal.slice(position + 4, partTotal.length);
+      let partHeaders = { };
+      let status = 500;
+      let lineNumber = 0;
+
+      // parse headers and status code
+
+      partHeader.split("\r\n").forEach( line => {
+        if (lineNumber === 0) {
+          position = line.search("HTTP/1.1 ");
+          // TOOD if (position === undefined) {
+          // TOOD   break;
+          // TOOD }
+          status = Number(line.slice(9, 9 + 3));
+        } else {
+          let splits = line.split(":");
+          partHeaders[splits[0].trim()] = splits[1].trim();
+        }
+        lineNumber = lineNumber + 1;
+      });
+      let part = { headers: partHeaders, body: partBody, status: status, contentId: contentId };
+      this.parts.push(part);
+      body = body.slice(nextBoundary, body.length);
+    }
+
+    if (!valid) {
+      throw Error("invalid multipart response received");
+    }
+
+    return this.parts;
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////;
+  // get the string representation of a multipart message body;
+  ////////////////////////////////////////////////////////////////////////////////;
+
+  stringify() {
+    let body = "";
+    this.parts.forEach(part => {
+      // boundary;
+      body += "--" + this.boundary + "\r\n";
+      // header;
+      body += "Content-Type: application/x-arango-batchpart";
+
+      if (part.contentId !== "") {
+        body += "\r\nContent-Id: " + part.contentId;
+      }
+
+      body += "\r\n\r\n";
+
+      // inner header;
+      body += part.method + " " + part.url + " HTTP/1.1\r\n";
+
+      for (const [key, value] of Object.entries(part.headers)) {
+        body += key + ": " + value + "\r\n";
+      }
+      // header/body separator;
+      body += "\r\n";
+      
+      // body;
+      body += part.body + "\r\n";
+    });
+    body += "--" + this.boundary + "--\r\n";
+    return body;
+  }
+};
+
+
+function using_disallowed_methodsSuite () {
+  return {
+
+    test_checks_whether_GET_is_allowed_on___api_batch: function() {
+      let cmd = "/_api/batch";
+      let doc = arango.GET_RAW(cmd);
+
+      assertEqual(doc.code, 405);
+      assertTrue(doc.parsedBody['error']);
+      assertEqual(doc.parsedBody['code'], 405);
+    },
+
+    test_checks_whether_HEAD_is_allowed_on___api_batch: function() {
+      let cmd = "/_api/batch";
+      let doc = arango.HEAD_RAW(cmd);
+
+      assertEqual(doc.code, 405);
+      assertEqual(doc.body, undefined);
```

Comment:
nit: can be simplified...
```suggestion
      assertUndefined(doc.body);
```